src/Storages/StorageReplicatedMergeTree.cpp

```diff
@@ -2776,7 +2787,37 @@ void StorageReplicatedMergeTree::cloneReplica(const String & source_replica, Coo
     for (const String & name : active_parts)
     {
         LogEntry log_entry;
-        log_entry.type = LogEntry::GET_PART;
+
+        if (!are_restoring_replica)
+            log_entry.type = LogEntry::GET_PART;
+        else
```

Comment:
```
if (condition)
    do_something();
else
    do_something_else();
```

is usually more readable than

```
if (!condition)
    do_something_else();
else
    do_something();
```

Reply:
Sure, in your example. But I tend to place branches with less code in the top, so it's more like

```cpp
if (!condition)
  do_stuff();
else
{
  do_a_lot_of_stuff();
  do_something_else();
}
```